product-accelerators/apim/components/org.wso2.healthcare.apim.backendauth/src/main/java/org/wso2/healthcare/apim/backendauth/PrivateKeyJWTBackendAuthenticator.java

```diff
@@ -0,0 +1,313 @@
+
+        log.info("PrivateKeyJWTBackendAuthenticator mediator is started.");
+
+        String accessToken;
+        String tokenEndpoint = backendAuthConfig.getAuthEndpoint();
+        String keyAlias = backendAuthConfig.getPrivateKeyAlias();
+        char[] keyStorePass = CarbonUtils.getServerConfiguration().getFirstProperty(
+                "Security.KeyStore.Password").toCharArray();
+        char[] trustStorePass = CarbonUtils.getServerConfiguration().getFirstProperty(
+                "Security.TrustStore.Password").toCharArray();
+        String clientId = backendAuthConfig.getClientId();
+        int tenantId = Integer.parseInt(messageContext.getProperty(Constants.TENANT_INFO_ID).toString());
+        KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(tenantId);
+        Key privateKey;
+        Certificate publicCert;
+
+        if (log.isDebugEnabled()) {
+            log.debug("Configured client ID: " + clientId);
+        }
+
+        try {
+            KeyStore primarykeyStore = keyStoreManager.getPrimaryKeyStore();
+            privateKey = primarykeyStore.getKey(keyAlias, keyStorePass);
+            publicCert = keyStoreManager.getDefaultPrimaryCertificate();
+        } catch (Exception e) {
+            log.error("Error occurred while retrieving private key from keystore.", e);
+            throw new OpenHealthcareRuntimeException(e);
+        }
+
+        Token token = TokenManager.getToken(clientId, tokenEndpoint);
+        if (token == null || !token.isActive()) {
+            if (log.isDebugEnabled()) {
+                log.debug("Access token not available in TokenManager.");
+            }
+            try {
+                token = getAndAddNewToken(clientId, tokenEndpoint, messageContext, privateKey, publicCert, trustStorePass);
+            } catch (OpenHealthcareException e) {
+                log.error("Error occurred while retrieving access token.",e);
+                throw new OpenHealthcareRuntimeException(e);
+            }
+        }
+        accessToken = token.getAccessToken();
+
+        messageContext.setProperty(Constants.PROPERTY_ACCESS_TOKEN, accessToken);
+
+        if (messageContext instanceof Axis2MessageContext) {
+            org.apache.axis2.context.MessageContext axisMsgCtx =
+                    ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+            Object headers = axisMsgCtx.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+            if (headers instanceof Map) {
+                Map headersMap = (Map) headers;
+                if (headersMap.get(Constants.HEADER_NAME_AUTHORIZATION) != null) {
+                    headersMap.remove(Constants.HEADER_NAME_AUTHORIZATION);
+                }
+                headersMap.put(Constants.HEADER_NAME_AUTHORIZATION, Constants.HEADER_VALUE_BEARER + accessToken);
+                axisMsgCtx.setProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS, headersMap);
+            }else {
```

Comment:
Formatting issues

Reply:
fixed in [b8da50d](https://github.com/wso2/healthcare-accelerator/pull/7/commits/b8da50dbe832d38ee8259489c5c03798ba7ca81d)